packages/common/src/internal/clients/cache/IDataClient.ts

```diff
@@ -0,0 +1,11 @@
+import {CacheGet, CacheSet} from '../../../index';
+
+export interface IDataClient {
+  get(cacheName: string, key: string | Uint8Array): Promise<CacheGet.Response>;
+  set(
+    cacheName: string,
+    key: string | Uint8Array,
+    value: string | Uint8Array,
+    ttl?: number
```

Comment:
im not sure if we want the internal data clients to also use the custom option types `SetOptions`? They arent currently, but then we wouldnt need separate interfaces for the public data clients and internal data clients

Reply:
I am inclined to leave it as is for now, and then maybe we can come back it it once we merge in the feature branch to main

Reply:
Can loop back to this later